src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -730,5 +783,31 @@ public ExtendedFilterResult(VSCompletionItem item, FilterResult filterResult)
                 FilterResult = filterResult;
             }
         }
+
+        private class NullablePatternMatchComparer : IComparer<ExtendedFilterResult>
```

Comment:
i don't get this name.  Also not sure why we need an ExtendedFilterResult.  Why not merge it all into one FilterResult struct?

Reply:
The name is outdated, should be `ExtendedFilterResultComparer` now. I forgot to change it. I think the reason for having an `ExtendedFilterResult` is because `FilterResult` is in Feature layer so can't have reference to editor.